### PR TITLE
Move interfaces and enum to Abstractions

### PR DESCRIPTION
**Pull request by fileman**
Move interfaces and enum to new Microsoft.AspNetCore.Datasync.Abstractions nuget to use it in DDD solutions, without Microsoft.AspNetCore.* references into Models projects.

### CODE REVIEW

**Comment by adrianhall** [comment]
Hi there @fileman - I'm not inclined to accept this one.  I keep the abstractions in the core package because you need to provide the core package as a dependency for all repositories.  Adding another package doesn't feel right in this case.

Can you provide more details on the problem you are seeing without the abstractions?

**Comment by fileman** [comment]
Hi @adrianhall , 
the core package has many references to other packages and requires .net 5.0, instead often the models are defined in a .netstandard class library, which could be shared in other projects or at least that's my case.

As example I'm working to integrate the new Datasync into ABP.IO framework, where models are in Domain project which is a .netstandard class library that can be shared in MVC, Blazor Server / WASM, etc...

**Comment by adrianhall** [comment]
Interesting concept.  Let me take a look at this for beta-9 once I'm through with issue #255 this week.  I'm all for supporting inclusion in other platforms and I hadn't even thought about a separate domain project. 

If you can, please open an issue for tracking this, then mention the PR as a solution.  I've got some more questions around why you use .net standard rather than .net 6.0 projects - a PR isn't a good place for those discussions.

**Review by adrianhall — COMMENTED** [review]
Do we want to add Abstractions as a project reference to the EFCore and InMemory projects, instead of relying on the transitive dependency in Datasync?

File: sdk/dotnet/src/Microsoft.AspNetCore.Datasync.Abstractions/Microsoft.AspNetCore.Datasync.Abstractions.csproj [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	  <LangVersion>9.0</LangVersion>
+	  <NeutralLanguage>en</NeutralLanguage>
+	  <PackageId>Microsoft.AspNetCore.Datasync.Abstractions</PackageId>
+	  <PackageLicenseExpression>MIT</PackageLicenseExpression>
+	  <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+	  <PackageTags>microsoft aspnetcore cloud datasync webapi</PackageTags>
+	  <ProductName>Microsoft.AspNetCore.Datasync.Abstractions</ProductName>
+	  <ProjectUrl>https://github.com/Azure/azure-mobile-apps</ProjectUrl>
+	  <RepositoryUrl>https://github.com/Azure/azure-mobile-apps</RepositoryUrl>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (adrianhall) [inline sdk/dotnet/src/Microsoft.AspNetCore.Datasync.Abstractions/Microsoft.AspNetCore.Datasync.Abstractions.csproj:19]: Needs to be net6.0 (to match the PR for #255)

**Review by adrianhall — CHANGES_REQUESTED** [review]

File: sdk/dotnet/src/Microsoft.AspNetCore.Datasync.Abstractions/Microsoft.AspNetCore.Datasync.Abstractions.csproj (hunk 1 above)
Comment (fileman) [reply sdk/dotnet/src/Microsoft.AspNetCore.Datasync.Abstractions/Microsoft.AspNetCore.Datasync.Abstractions.csproj]: you haven't requested that, but I have updated even language version

**Review by adrianhall — APPROVED** [review]
Love it - think it's ready.  Pause on merge while we get the new test infra, net6.0 updates, and #255 fixed, though.  

Thanks for the contribution!

**Comment by scout208** [comment]
Is .NET Standard still relevant with the release of .NET 5? From what I understand all of the platforms you are interested in supporting can work with .NET 5.

**Comment by fileman** [comment]
> Is .NET Standard still relevant with the release of .NET 5? From what I understand all of the platforms you are interested in supporting can work with .NET 5.

https://github.com/Azure/azure-mobile-apps/issues/258#issuecomment-970731332

**Comment by scout208** [comment]
> > Is .NET Standard still relevant with the release of .NET 5? From what I understand all of the platforms you are interested in supporting can work with .NET 5.
> 
> [#258 (comment)](https://github.com/Azure/azure-mobile-apps/issues/258#issuecomment-970731332)

Okay, thanks. I hadn't seen that issue yet.

**Pull request merged**